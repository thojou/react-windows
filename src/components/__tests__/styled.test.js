import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import {
  Area, Button,
  Grid,
  TitleArea,
  WindowBody,
  WindowFullscreen,
  WindowToolbar,
  WindowToolbarWrapper,
  WindowWrapper
} from '../styled';

describe('WindowWrapper', () => {
  it('should render', () => {
    const tree = renderer.create(<WindowWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render on focus', () => {
    const tree = renderer.create(<WindowWrapper focused={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('WindowBody', () => {
  it('should render', () => {
    const tree = renderer.create(<WindowBody />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('WindowFullscreen', () => {
  it('should render', () => {
    const tree = renderer.create(<WindowFullscreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('WindowToolbarWrapper', () => {
  it('should render', () => {
    const tree = renderer.create(<WindowToolbarWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('WindowToolbar', () => {
  it('should render', () => {
    const tree = renderer.create(<WindowToolbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Grid', () => {
  it('should render', () => {
    const tree = renderer.create(<Grid />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should minimize', () => {
    const tree = renderer.create(<Grid minimized={true} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('border', 'none');
    expect(tree).toHaveStyleRule('max-width', '225px');
  });

  it('should accept areas', () => {
    const areas = "'brand header' 'sidebar body'";
    const tree = renderer.create(<Grid areas={areas} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('grid-template-areas', areas);
  });

  it('should accept rows', () => {
    const rows = '1fr auto';
    const tree = renderer.create(<Grid rows={rows} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('grid-template-rows', rows);
  });

  it('should accept columns', () => {
    const columns = '1fr auto';
    const tree = renderer.create(<Grid columns={columns} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('grid-template-columns', columns);
  });
});

describe('Area', () => {
  it('should render', () => {
    const tree = renderer.create(<Area />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should accept name', () => {
    const name = 'header';
    const tree = renderer.create(<Area name={name} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('grid-area', name);
  });
});

describe('TitleArea', () => {
  it('should render', () => {
    const tree = renderer.create(<TitleArea />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button', () => {
  it('should render', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
