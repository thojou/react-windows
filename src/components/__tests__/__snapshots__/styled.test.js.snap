// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area should accept name 1`] = `
.c0 {
  height: 100%;
  grid-area: header;
  box-sizing: border-box;
}

.c0 > * {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
  name="header"
/>
`;

exports[`Area should render 1`] = `
.c0 {
  height: 100%;
  grid-area: default;
  box-sizing: border-box;
}

.c0 > * {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
  name="default"
/>
`;

exports[`Button should render 1`] = `
.c0 {
  border: none;
  background-color: transparent;
  color: #666666;
  cursor: pointer;
}

.c0:hover {
  color: #A7893D;
}

.c0:last-child {
  padding-right: 0;
}

<button
  className="c0"
/>
`;

exports[`Grid should accept areas 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  display: grid;
  color: #666666;
  font: normal bold 16px/21px "Segoe UI";
  background-color: #ffffff;
  border: 1px solid #E8E8E8;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  grid-template-areas: 'brand header' 'sidebar body';
}

<div
  className="c0"
  rows="1fr"
/>
`;

exports[`Grid should accept columns 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  display: grid;
  color: #666666;
  font: normal bold 16px/21px "Segoe UI";
  background-color: #ffffff;
  border: 1px solid #E8E8E8;
  grid-template-columns: 1fr auto;
  grid-template-rows: 1fr;
  grid-template-areas: default;
}

<div
  className="c0"
  rows="1fr"
/>
`;

exports[`Grid should accept rows 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  display: grid;
  color: #666666;
  font: normal bold 16px/21px "Segoe UI";
  background-color: #ffffff;
  border: 1px solid #E8E8E8;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr auto;
  grid-template-areas: default;
}

<div
  className="c0"
  rows="1fr auto"
/>
`;

exports[`Grid should minimize 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  display: grid;
  color: #666666;
  font: normal bold 16px/21px "Segoe UI";
  background-color: #ffffff;
  border: 1px solid #E8E8E8;
  max-width: 225px;
  box-shadow: none;
  border: none;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  grid-template-areas: default;
}

<div
  className="c0"
  rows="1fr"
/>
`;

exports[`Grid should render 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  display: grid;
  color: #666666;
  font: normal bold 16px/21px "Segoe UI";
  background-color: #ffffff;
  border: 1px solid #E8E8E8;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  grid-template-areas: default;
}

<div
  className="c0"
  rows="1fr"
/>
`;

exports[`TitleArea should render 1`] = `
.c0 {
  height: 100%;
  grid-area: default;
  box-sizing: border-box;
  padding: 0.5rem 0.938rem;
  background-color: #E8E8E8;
}

.c0 > * {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
  name="default"
/>
`;

exports[`WindowBody should render 1`] = `
.c0 {
  display: block;
  height: 100%;
}

<div
  className="c0"
/>
`;

exports[`WindowFullscreen should render 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

<div
  className="c0"
/>
`;

exports[`WindowToolbar should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
}

<div
  className="c0"
/>
`;

exports[`WindowToolbarWrapper should render 1`] = `
.c0 {
  width: calc(100% - 260px);
  background-color: red;
  position: absolute;
  left: 260px;
  bottom: 0;
}

<div
  className="c0"
/>
`;

exports[`WindowWrapper should render 1`] = `
.c0 {
  overflow: hidden;
  height: 100%;
  box-shadow: 0 3px 6px #E8E8E8;
}

<div
  className="c0"
/>
`;

exports[`WindowWrapper should render on focus 1`] = `
.c0 {
  overflow: hidden;
  height: 100%;
  box-shadow: 0 3px 6px #A7893D;
}

<div
  className="c0"
/>
`;
